internal/utils/utils.go

```diff
@@ -40,7 +40,8 @@ func GetArchitecture() string {
 // DownloadFile will download a url and store it in a temporary path.
 // It writes to the destination file as it downloads it, without
 // loading the entire file into memory.
-func DownloadFile(url string) (string, error) {
+func DownloadFile(url string, downloadPath string) (string, error) {
```

Comment:
I'm looking at the implementation and it's not clear to me if we need to return the path again, as it's passed in the download path. I can picture a new struct (i.e. `DownloadRequest`) with two attributes: url and downloadPath. The DownloadFile method would receive a pointer to an instance of this struct. If the downloadPath is empty, then create a temporary file (as it does now), populating the download path, otherwise use the value in the request. The method would still return the request's download path.

Wdyt?

Reply:
Yea if we're already passing in the downloadPath then a struct containing that information could be easily reused/referenced elsewhere.  I also tend to go with defined structs as it improves readability and intent